### DISCUSSION TITLE
Panel extension API improvements

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#227] [Q&A]
We want to prioritize enabling users to create extension panels that are as full-featured as first-party panels.

### Blocking issues

- [ ] Reset or seek event when playback loops, user seeks, data source is changed, etc
- [ ] Support common UI widgets in extension API (e.g. message path input)
- [ ] Support panel settings / layer editor in extension API
- [ ] Theme support
- [ ] Access Studio web fonts

### Nice to have

- [ ] Storybook in the template project
- [ ] Host panels inside iframes to better control error reporting
- [ ] Web workers in extensions
- [ ] WebAssembly support
- [ ] Migrate remaining panels to use extension API and deprecate old Panel API (forcing function)

**Comment by defunctzombie** [comment]
Removing this from the roadmap project. Right now this tracks a set of high level ideas. When ready we need to scope in specific items.

@amacneil Maybe we should close this instead and scope specific improvements like we have with "message converter" extension.
